### PR TITLE
Got rid of assigning the window.onerror property (#1295)

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

File: src/client/sandbox/node/window.js [hunk 1]
```diff
@@ -102,11 +101,22 @@ export default class WindowSandbox extends SandboxBase {
...
+        if (event.defaultPrevented)
+            return;
+
+        if (event.type === 'unhandledrejection') {
             const reason = WindowSandbox._formatUnhandledRejectionReason(event.reason);
 
             this._raiseUncaughtJsErrorEvent(this.UNHANDLED_REJECTION_EVENT, reason, this.window);
         }
+        else if (event.type === 'error') {
+            const message = event.message;
```
Comment (miherlosev) [inline src/client/sandbox/node/window.js]: Don't need a separate variable `message`

File: test/client/fixtures/sandbox/node/window-test.js [hunk 2]
```diff
@@ -351,6 +357,57 @@ test('window.onerror must be overriden (B238830)', function () {
     ok(!error);
 });
 
+test('should not rise hh event when the error event prevented', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/window-test.js]: `should not raise internal events if an origin error event is prevented`

File: test/client/fixtures/sandbox/node/window-test.js [hunk 3]
```diff
@@ -351,6 +357,57 @@ test('window.onerror must be overriden (B238830)', function () {
...
+    var testPreventing = function () {
+        return new hammerhead.Promise(function (resolve) {
+            var testTimeout = null;
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/window-test.js]: `var timeoutId = ...`

File: test/client/fixtures/sandbox/node/window-test.js [hunk 4]
```diff
@@ -351,6 +357,57 @@ test('window.onerror must be overriden (B238830)', function () {
...
+
+            var onUncaughtError = function () {
+                clearTimeout(testTimeout);
+                hammerhead.off(hammerhead.EVENTS.uncaughtJsError, onUncaughtError);
+                ok(false, 'hh event not prevented');
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/window-test.js]: `internal event is raised`
`internal event is not raised`

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**